go.mod

```diff
@@ -3,9 +3,57 @@ module google.golang.org/genproto
 go 1.19
 
 require (
+	cloud.google.com/go/analytics v0.11.0
+	cloud.google.com/go/area120 v0.5.0
+	cloud.google.com/go/artifactregistry v1.6.0
 	cloud.google.com/go/asset v1.5.0
+	cloud.google.com/go/assuredworkloads v1.5.0
+	cloud.google.com/go/automl v1.5.0
+	cloud.google.com/go/billing v1.4.0
+	cloud.google.com/go/binaryauthorization v1.1.0
+	cloud.google.com/go/cloudtasks v1.5.0
+	cloud.google.com/go/datacatalog v1.5.0
+	cloud.google.com/go/dataflow v0.6.0
+	cloud.google.com/go/dataform v0.3.0
+	cloud.google.com/go/datalabeling v0.5.0
+	cloud.google.com/go/dataqna v0.5.0
+	cloud.google.com/go/datastream v1.2.0
+	cloud.google.com/go/dialogflow v1.15.0
+	cloud.google.com/go/documentai v1.7.0
+	cloud.google.com/go/domains v0.6.0
 	cloud.google.com/go/edgecontainer v0.1.0
-	cloud.google.com/go/security v1.5.0
+	cloud.google.com/go/functions v1.6.0
+	cloud.google.com/go/gaming v1.5.0
+	cloud.google.com/go/gkeconnect v0.5.0
+	cloud.google.com/go/gkehub v0.9.0
+	cloud.google.com/go/language v1.4.0
+	cloud.google.com/go/lifesciences v0.5.0
+	cloud.google.com/go/mediatranslation v0.5.0
+	cloud.google.com/go/memcache v1.4.0
+	cloud.google.com/go/metastore v1.5.0
+	cloud.google.com/go/networkconnectivity v1.4.0
+	cloud.google.com/go/networksecurity v0.5.0
+	cloud.google.com/go/notebooks v1.2.0
+	cloud.google.com/go/osconfig v1.7.0
+	cloud.google.com/go/oslogin v1.4.0
+	cloud.google.com/go/phishingprotection v0.5.0
+	cloud.google.com/go/privatecatalog v0.5.0
+	cloud.google.com/go/recaptchaenterprise/v2 v2.1.0
+	cloud.google.com/go/recommendationengine v0.5.0
+	cloud.google.com/go/recommender v1.5.0
+	cloud.google.com/go/redis v1.7.0
+	cloud.google.com/go/retail v1.8.0
+	cloud.google.com/go/scheduler v1.4.0
+	cloud.google.com/go/secretmanager v1.6.0
+	cloud.google.com/go/security v1.7.0
+	cloud.google.com/go/securitycenter v1.13.0
+	cloud.google.com/go/servicedirectory v1.4.0
+	cloud.google.com/go/speech v1.6.0
+	cloud.google.com/go/talent v1.1.0
+	cloud.google.com/go/videointelligence v1.6.0
+	cloud.google.com/go/vision/v2 v2.2.0
+	cloud.google.com/go/webrisk v1.4.0
+	cloud.google.com/go/workflows v1.6.0
```

Comment:
this is _really_ unfortunate... this library is referenced by a ton of generic API libraries, and adding all of these propagates these requires to the dependency graphs of all consumers. for example, when Kubernetes is [attempting to update to a version that includes this change](https://github.com/kubernetes/kubernetes/pull/113225/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6), it added 53 dependencies to our graph

was that impact considered? is there any suggested mitigation?

Reply:
You are right, this migration we are doing does add a bunch of dependencies to this package. But in reality my assumption is that most of these will be indirect for most users and never need to actually be pulled in -- for example if you ran `go mod vendor` does a lot more code actually get pulled in?

What packages, or tree of packages does Kubernetes pull in? Depending on that there may be some tricks we can do to lessen the burden.

Reply:
they don't get vendored, but all the cloud.google.com/go/* modules get added to our dependency graph and passed along to everything downstream of kubernetes as well

opened https://github.com/kubernetes/kubernetes/issues/113366 to capture the genproto packages we're actually using and start to work through possible ways to insulate consumers of those generic packages from unused cloud dependencies

Reply:
SG, resolving this and we can use that issue for discussion.